covid19_sfbayarea/data/marin_scraper.py

```diff
@@ -0,0 +1,303 @@
+
+	driver.switch_to.frame(frame)
+	# Grab the raw data out of the link's href attribute
+	csv_data = driver.find_element_by_class_name('dw-data-link').get_attribute('href')
+	# Switch back to the parent frame to "reset" the context
+	driver.switch_to.parent_frame()
+	
+	# Deal with the data
+	if csv_data.startswith('data:'):
+		media, data = csv_data[5:].split(',', 1)
+		# Will likely always have this kind of data type
+		if media != 'application/octet-stream;charset=utf-8':
+			raise ValueError(f'Cannot handle media type "{media}"')
+		csv_string = unquote_plus(data)
+
+	# Then leave the iframe
+	driver.switch_to.default_content()
+
+	return csv_string
+
+def get_metadata(url: str, chart_ids: str) -> Tuple:
+	notes = []
+	driver = webdriver.Chrome('/Users/angelakwon/Downloads/chromedriver') # change this to point to Github one
+	driver.implicitly_wait(30)
+	driver.get(url)
+	soup = BeautifulSoup(driver.page_source, 'html5lib')
+	metadata = []
+
+	to_be_matched = ['Total Cases, Recovered, Hospitalizations and Deaths by Date Reported', 'Daily Count of Positive Results and Total Tests for Marin County Residents by Test Date ', 'Cases, Hospitalizations, and Deaths by Age, Gender and Race/Ethnicity ']
+	chart_metadata = []
+
+	for text in to_be_matched:
+		target = soup.find('h4',text=text)
+		if not target:
+			raise ValueError('Cannot handle this header.')
+		for sib in target.find_next_siblings()[:1]: # I only want the first paragraph tag
+			# Is it more efficient to use something like (soup object).select('h1 + p') to grab the first paragraph that follows?
+			metadata += [sib.text]
+
+	# Metadata for each chart visualizing the data of the csv file I'll pull. There's probably a better way to organize this.
+	for chart_id in chart_ids.values():
+		frame = driver.find_element_by_css_selector(f'iframe[src*="//datawrapper.dwcdn.net/{chart_id}/"]')
+		driver.switch_to.frame(frame)
+		# The metadata for the charts is located in elements with the class `dw-chart-notes' 
+		for c in driver.find_elements_by_class_name('dw-chart-notes'):
+			chart_metadata.append(c.text)
+
+		# Switch back to the parent frame to "reset" the context
+		driver.switch_to.parent_frame()
+
+	driver.quit() 
+
+	# Return the metadata. I take the set of the chart_metadata since there are repeating metadata strings.
+	return metadata, list(set(chart_metadata)) 
+
+def get_case_series(chart_id: str, url: str) -> List:
+	"""This method extracts the date, number of cumulative cases, and new cases."""
+	csv_ = extract_csvs(chart_id, url)
+	series = []
+
+	csv_strs = csv_.splitlines()
+	keys = csv_strs[0].split(',')
+	
+	if keys != ['Date', 'Total Cases', 'Total Recovered*', 'Total Hospitalized', 'Total Deaths']:
+		raise ValueError('The headers have changed')
+
+	case_history = []
+
+	for row in csv_strs[1:]:
+	 	daily = {}
+	 	# Grab the date in the first column
+	 	date_time_obj = datetime.strptime(row.split(',')[0], '%m/%d/%Y')
+	 	daily["date"] = date_time_obj.isoformat()
```

Comment:
I think this is supposed to be a `date`, not a `datetime`, right?

Reply:
Yes, that variable should be date_obj! Unless that's not what you're referring to?

Reply:
No, sorry, I wasn’t clear. I was referring to the fact that `datetime.strptime()` returns an instance of `datetime.datetime` rather than `datetime.date`. And calling `.isoformat()` on a `datetime` gets you a string like `2020-05-12T00:00:00`, but we want a string like `2020-05-12`.

So if `row.split(',')[0]` was `'5/12/2020'` in this code, you’d get:

```py
>>> date_time_obj = datetime.strptime('5/12/2020', '%m/%d/%Y')
>>> date_time_obj.isoformat()
'2020-05-12T00:00:00'
```

You can fix that in a couple ways:

```py
# Convert to a datetime.date first:
date_time_obj.date().isoformat()

# Or provide an explicit format instead of isoformat:
date_time_obj.strftime('%Y-%m-%d')
```

Reply:
ah my bad - for whatever reason I thought we wanted a string like `2020-05-12T00:00:00`. I'll make sure to convert to a date time.date first, thanks Rob!